### PR TITLE
Add thermal throttling check for H100 GPU systems

### PR DESCRIPTION
**Pull request by moraleslazaro**
## Summary

This PR implements a new diagnostic test `thermal_throttling_check` for the BM.GPU.H100.8 shape. The test detects GPU thermal throttling events using nvidia-smi to monitor various throttling conditions and GPU temperatures.

## Changes

- Added Python script `/scripts/BM.GPU.H100.8/thermal_throttling_check.py`
- Added Bash script `/scripts/BM.GPU.H100.8/thermal_throttling_check.sh`
- Added Go implementation in `/internal/level1_tests/thermal_throttling_check.go`
- Added test file `/internal/level1_tests/thermal_throttling_check_test.go`
- Updated test configuration in `/internal/test_limits/test_limits.json`
- Updated recommendations in `/configs/recommendations.json`
- Enhanced executor module with `RunNvidiaSMIThermalThrottlingQuery()` function
- Updated reporter module to support thermal throttling test results
- Added support for table, friendly, and JSON output formats

## Test Implementation

The test uses `nvidia-smi` to query thermal throttling information:
- Monitors hardware slowdown, thermal slowdown, power brake slowdown, and software thermal slowdown
- Checks GPU and memory temperatures
- Reports failures with specific GPU information and throttling reasons
- Integrates with existing reporter and recommender systems

## Output Formats

The test supports all output formats with proper display:

### Table Format
```
┌─────────────────────────────────────────────────────────────────┐
│                    DIAGNOSTIC TEST RESULTS                      │
├─────────────────────────────────────────────────────────────────┤
│ TEST NAME              │ STATUS  │ DETAILS                      │
├─────────────────────────────────────────────────────────────────┤
│ Thermal Throttling     │ ✅      │ No Thermal Issues            │
└─────────────────────────────────────────────────────────────────┘
```

### Friendly Format
```
🌡️ GPU Thermal Throttling Check
   ------------------------------
   ✅ GPU Thermal Throttling Check: No thermal throttling detected across all GPUs (PASSED)
```

### JSON Format
- Machine-readable structured output with detailed throttling data

## Configuration
- **Enabled**: BM.GPU.H100.8 (H100 systems)
- **Disabled**: BM.GPU.B200.8, BM.GPU.GB200.4 (not applicable)
- **Test Category**: LEVEL_1
- **Fault Code**: HPCGPU-0014-0001

## Test Plan

- [ ] Verify all existing tests continue to pass
- [ ] Confirm new test runs successfully on BM.GPU.H100.8 instances
- [ ] Validate Python and Bash scripts produce expected output
- [ ] Test error conditions and edge cases
- [ ] Verify output formatting across all report modes
- [ ] Confirm cross-platform compilation with `make all-cross`

### CODE REVIEW

File: internal/executor/nvidia_smi.go [hunk 1]
```diff
@@ -469,4 +469,45 @@ func truncateString(s string, maxLen int) string {
 		return s
 	}
 	return s[:maxLen] + "..."
+}
+
+// RunNvidiaSMIThermalThrottlingQuery runs nvidia-smi command to query thermal throttling status
```
Comment (bharatrajan) [inline internal/executor/nvidia_smi.go:474]: Shall we utilize `RunNvidiaSMIQuery` function from here https://github.com/ai2-compute-gpu/oci-dr-hpc-v2/blob/main/internal/executor/nvidia_smi.go#L69C6-L69C23

**Review by bharatrajan — APPROVED** [review]

**Review by bobrbooth — APPROVED** [review]

File: internal/executor/nvidia_smi.go (hunk 1 above)
Comment (moraleslazaro) [reply internal/executor/nvidia_smi.go]: Thanks @bharatrajan. Good suggestion, went ahead and used [RunNvidiaSMIQueryWithHeaders](https://github.com/ai2-compute-gpu/oci-dr-hpc-v2/pull/27/files#diff-c1b40c80182b915c629ebe1be22d209f43832f18a6ac29e6710862fc00c160cf) since the original [RunNvidiaSMIQuery](https://github.com/ai2-compute-gpu/oci-dr-hpc-v2/blob/main/internal/executor/nvidia_smi.go#L69-L105) doesn't include the headers.
